packages/theme/src/cli/commands/theme/metafields/pull.ts

```diff
@@ -0,0 +1,44 @@
+import {themeFlags} from '../../../flags.js'
+import {metafieldsPull, MetafieldsPullFlags} from '../../../services/metafields-pull.js'
+import ThemeCommand from '../../../utilities/theme-command.js'
+import {globalFlags} from '@shopify/cli-kit/node/cli'
+import {Flags} from '@oclif/core'
+
+export default class MetafieldsPull extends ThemeCommand {
+  static summary = 'Download metafields definitions from your shop into a local file.'
+
+  static descriptionWithMarkdown = `Retrieves metafields from Shopify Admin.
+
+If the metafields file already exists, it will be overwritten.`
+
+  static description = this.descriptionWithoutMarkdown()
+
+  static flags = {
+    ...globalFlags,
```

Comment:
We initially couldn't support theme access app (and environments) because of permission issues, but now that we have that resolved, i can add it back 👍